gqcp/include/ONVBasis/SpinResolvedSelectedONVBasis.hpp

```diff
@@ -378,7 +395,7 @@ class SpinResolvedSelectedONVBasis {
                 SpinUnresolvedONV beta_J = onv_J.onv(Spin::beta);
 
                 // 1 excitation in the alpha part, 0 in the beta part.
-                if ((alpha_I.countNumberOfDifferences(alpha_J) == 2) && (beta_I.countNumberOfDifferences(beta_J) == 0)) {
+                if ((alpha_I.countNumberOfExcitations(alpha_J) == 1) && (beta_I.countNumberOfExcitations(beta_J) == 0)) {
```

Comment:
Thanks for cleaning this up. Not only did you fix a 'wrong use of an API', but it's clearer to read as well.